src/Elmish.WPF/ViewModel.fs

```diff
@@ -107,37 +107,38 @@ type internal TwoWayBinding<'model> = {
   Set: obj -> 'model -> unit
 }
 
-type internal SubModelBinding<'model, 'msg, 'bindingModel, 'bindingMsg> = {
-  SubModelData: SubModelData<'model, 'msg, 'bindingModel, 'bindingMsg>
-  Vm: ViewModel<'bindingModel, 'bindingMsg> voption ref
+type internal SubModelBinding<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel> = {
+  SubModelData: SubModelData<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel>
+  Vm: 'bindingViewModel voption ref
 }
 
-and internal SubModelWinBinding<'model, 'msg, 'bindingModel, 'bindingMsg> = {
-  SubModelWinData: SubModelWinData<'model, 'msg, 'bindingModel, 'bindingMsg>
+and internal SubModelWinBinding<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel> = {
+  SubModelWinData: SubModelWinData<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel>
   WinRef: WeakReference<Window>
   PreventClose: bool ref
-  VmWinState: WindowState<ViewModel<'bindingModel, 'bindingMsg>> ref
+  VmWinState: WindowState<'bindingViewModel> ref
 }
 
-and internal SubModelSeqUnkeyedBinding<'model, 'msg, 'bindingModel, 'bindingMsg> = {
-  SubModelSeqUnkeyedData: SubModelSeqUnkeyedData<'model, 'msg, 'bindingModel, 'bindingMsg>
-  Vms: ObservableCollection<ViewModel<'bindingModel, 'bindingMsg>>
+and internal SubModelSeqUnkeyedBinding<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel> = {
+  SubModelSeqUnkeyedData: SubModelSeqUnkeyedData<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel>
+  Vms: ObservableCollection<'bindingViewModel>
```

Comment:
Here (and similar) is why the `ObservableCollection<ViewModel<obj,obj>>` became `ObservableCollection<obj>`. Thanks to the updated box function, we now box the `'bindingViewModel -> obj` where previously we just boxed the `ViewModel<'bindingModel,'bindingMsg> -> ViewModel<obj,obj>`

Reply:
Similarly, that also means that we never have `ViewModel<obj,obj>` even in other places, because we either box them directly to `obj` or we have the original `ViewModel<'bindingModel,'bindingMsg>`.